internal/build/resolver/user_builds.go

```diff
@@ -12,57 +12,43 @@ import (
 	"golang.org/x/sync/errgroup"
 )
 
-// ResolveBuildFromCurrentBranch Finds the most recent build for the branch in the current working directory
-func ResolveBuildForCurrentUser(branch string, pipelineResolver pipelineResolver.PipelineResolverFn, f *factory.Factory) BuildResolverFn {
-	return func(ctx context.Context) (*build.Build, error) {
-		var pipeline *pipeline.Pipeline
-		var user *buildkite.User
+// ResolveBuildForUser Finds the most recent build for the user based on the provided options
+func ResolveBuildForUser(ctx context.Context, userInfo string, opt *buildkite.BuildsListOptions, pipelineResolver pipelineResolver.PipelineResolverFn, f *factory.Factory) (*build.Build, error) {
```

Comment:
I think it might be better if we didn't use the `opt *buildkite.BuildsListOptions` in the argument list to avoid exposing the underlying API options. It looks like we could change this argument to the branch string and then create the BuildsListOptions within this function